### PR TITLE
[SW-449] Support for Sparse Vectors and Upgrade to H2O 3.14.0.2

### PR DESCRIPTION
**Pull request by jakubhava**
SparkDataFrameConverter refactored. Now we first create flat dataframe in spark to simplify handling on sw side. 

The refactoring was needed in order to separate handling of vectors.

An improvement of this solution would be to drop the flag whether the vector is sparse or dense and make it part of the expected types

This change requires https://github.com/h2oai/h2o-3/pull/1348

### CODE REVIEW

**Comment by jakubhava** [comment]
Still some tests are failing on external cluster, but getting to the final state. Benchmarks shows this is a good step

**Comment by jakubhava** [comment]
Ready for review, this is final state

**Comment by jakubhava** [comment]
@mmalohlava `continuous-integration/jenkins/pr-head` shouldn't succeed as this change depends on different H2O version, even during compilation

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala [hunk 1]
```diff
@@ -42,4 +42,37 @@ private[external] trait ExternalBackendUtils extends SharedBackendUtils{
...
+        ExternalFrameUtils.EXPECTED_LONG
+      } else if (clazz == classOf[java.lang.Float]) {
+        ExternalFrameUtils.EXPECTED_FLOAT
+      } else if (clazz == classOf[java.lang.Double]) {
+        ExternalFrameUtils.EXPECTED_DOUBLE
+      } else if (clazz == classOf[java.lang.String]) {
+        ExternalFrameUtils.EXPECTED_STRING
+      } else if (clazz == classOf[java.sql.Timestamp]) {
+        ExternalFrameUtils.EXPECTED_TIMESTAMP
+      } else if (clazz == classOf[org.apache.spark.mllib.linalg.Vector]) {
```
Comment (mmalohlava) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala]: For future: what about about `org.apache.spark.ml.linalg.Vector` ?

Comment (mmalohlava) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala]: Oki, found it, you are doing Vectors.toML(..) coversion before.

**Comment by mmalohlava** [comment]
Right now blocked by:
SW-516, SW-515

**Pull request merged**